src/SDK/Trace/ExporterFactory.php

```diff
@@ -133,12 +148,42 @@ private static function getOptionFromDsn(DsnInterface $dsn, string $parameter):
         return null;
     }
 
-    private static function buildExporter(string $protocol, string $endpoint = null, string $name = null, $args = null): SpanExporterInterface
+    private static function buildExporter($protocol, string $endpoint = null, string $name = null, $args = null): SpanExporterInterface
+    {
+        $exporterClass = self::KNOWN_EXPORTERS[self::normalizeProtocol($protocol)];
+        self::validateClass($exporterClass);
+
+        return call_user_func([$exporterClass, 'fromConnectionString'], $endpoint, $name, $args, $protocol);
+    }
+
+    /**
+     * @phan-suppress PhanTypeMismatchArgument
+     * @phan-suppress PhanUndeclaredClass
+     * @phan-suppress PhanParamTooMany
+     */
+    private static function buildExporterWithTransport(string $protocol): SpanExporterInterface
     {
         $exporterClass = self::KNOWN_EXPORTERS[self::normalizeProtocol($protocol)];
-        self::validateExporterClass($exporterClass);
+        self::validateClass($exporterClass);
+
+        $exporter = new $exporterClass(self::buildTransport($protocol), self::KNOWN_TRANSPORT_PROTOCOLS[$protocol]); // @phpstan-ignore-line
+        assert($exporter instanceof SpanExporterInterface);
+
+        return $exporter;
+    }
+
+    /**
+     * @phan-suppress PhanParamTooFew
+     * @psalm-suppress TooFewArguments
+     */
+    private static function buildTransport(string $protocol): TransportInterface
+    {
+        $factoryClass = self::KNOWN_TRANSPORT_FACTORIES[self::normalizeProtocol($protocol)];
+        self::validateClass($factoryClass);
+        $factory = new $factoryClass();
```

Comment:
Users should be able to provide custom transport factory implementations.

Reply:
I think we would need to invent a new env var for that, something like `OTEL_PHP_TRANSPORT_FACTORY_CLASS`...

Reply:
I'm thinking of a follow-up PR which allows registering exporter and transport factories, and using composer autoload->files to register our initial set.